backend/src/services/timeslotService.ts

```diff
@@ -61,6 +61,38 @@ const deleteById = async (id: number) => {
   await timeslot?.destroy();
 };
 
+const createPeriod = async (
+  period: { periodStart: Date, periodEnd: Date, name: string },
+  timeslot: { start: Date, end: Date },
+): Promise<Timeslot[]> => {
+  const oneWeek = 7 * 24 * 60 * 60 * 1000;
+  const { periodEnd } = period;
+  const timeslotGroup: { start: Date, end: Date }[] = [];
+  let { start, end } = timeslot;
+  start = new Date(start.getTime() + oneWeek);
+  end = new Date(end.getTime() + oneWeek);
+  while (end <= periodEnd) {
+    timeslotGroup.push({ start, end });
+    start = new Date(start.getTime() + oneWeek);
+    end = new Date(end.getTime() + oneWeek);
+  }
+  // eslint-disable-next-line no-restricted-syntax
+  for (const t of timeslotGroup) {
+    // eslint-disable-next-line no-await-in-loop
+    if (await timeslotsAreConsecutive(t)) {
+      throw new Error('Timeslot can\'t be consecutive');
+    }
+    // eslint-disable-next-line no-await-in-loop
+    const timeslots = await getInTimeRange(t.start, t.end);
+    if (timeslots.length > 0) {
+      throw new Error('Period has already timeslot');
+    }
+  }
```

Comment:
voisi välttää eslint jutut. Ehdotan että nostaa tuon tsconfigin targetin ja sitten:

```ts
const consecutivesFound = await Promise.all(
  timeslotGroup.map((ts) => timeslotsAreConsecutive(ts)),
);
if (consecutivesFound.some((found) => found === true)) {
  throw new Error("Timeslot can't be consecutive with another");
}

const overlaps = await Promise.all(
  timeslotGroup.map((ts) => getInTimeRange(ts.start, ts.end)),
);
if (overlaps.some((foundSlots) => foundSlots.length > 0)) {
  throw new Error('Period already has a timeslot');
}
```


Näin hautkin tapahtuu samaan aikaan eikä yksi kerrallaan, mikä on varmaan syy että miksi tuo eslint valittaakin

Reply:
Hieno ratkaisu. Korjattu käyttämään Promise.all.